apps/demo-game/src/reducers/ActionsReducer.ts

```diff
@@ -1,99 +1,91 @@
-import { Action } from '@gbl-uzh/platform'
+import { Action, ResultState } from '@gbl-uzh/platform'
 import { debugLog } from '@gbl-uzh/platform/dist/lib/util'
+import { PeriodFacts, PeriodSegmentFacts } from '@graphql/types'
 import { PrismaClient } from '@prisma/client'
-import { match } from 'ts-pattern'
+import { produce } from 'immer'
+import { P, match } from 'ts-pattern'
 
 export enum ActionTypes {
   DECIDE_BANK = 'DECIDE_BANK',
   DECIDE_BONDS = 'DECIDE_BONDS',
   DECIDE_STOCK = 'DECIDE_STOCK',
 }
 
+// TODO(Jakob):
+// - Ponder: Make input (state) and output (newState) always having consistent
+//   properties (with optional values) vs like it is now (each case could
+//   deliver a different output
+// - For the ActionReducer it is trivial and consistent
+//   (always the same properties as output)
+
+type PayloadType = {
+  playerArgs: {
+    decision: boolean
+  }
+  segmentFacts: PeriodSegmentFacts
+  periodFacts: PeriodFacts
+}
+
+type InputState = {
+  decisions: {
+    bank: boolean
+    bonds: boolean
+    stocks: boolean
+  }
+}
+
+type OutputState = InputState
+
 type Actions =
-  | Action<
-      ActionTypes.DECIDE_BANK,
-      {
-        playerArgs: {
-          decision: boolean
+  | Action<ActionTypes.DECIDE_BANK, PayloadType, PrismaClient>
+  | Action<ActionTypes.DECIDE_BONDS, PayloadType, PrismaClient>
+  | Action<ActionTypes.DECIDE_STOCK, PayloadType, PrismaClient>
+
+export function apply(
+  state: InputState,
+  action: Actions
+): ResultState<OutputState> {
+  // TODO: move this to platform? -> reducer should not have to care about isDirty and other non-user-logicstuff
+  const baseState: ResultState<OutputState> = {
+    result: state,
+    isDirty: false,
+  }
+
+  // TODO: the user reducer could just get the "draft" inside this function as first parameter
```

Comment:
will need to think about what the draft is in terms of type (not just an object)

Reply:
So far from what I have seen, its type should be the output type.